### ISSUE TITLE
Ajax calls with data

### ISSUE THREAD

**Issue by sabatesduran** [quark-dev/Phonon-Framework#100]
I have a problem when I send ajax request with ```data:```

```javascript
function getData(id) {
    var dataJson = {};
    dataJson.id = id;

    phonon.ajax({
        method: 'POST',
        url: 'http://url.com',
        dataType: 'json',
        data: dataJson,
        success: function(res, xhr) {
              // do something
            }
        },
        error: function(res, flagError, xhr) {
            console.error(flagError);
        }
    });
}

```

I test the webservice with Postman and it works correctly.
If I remove the ```data:``` (the webservice accepts no parameter) the call works.

I intent to debug the code but i don't find anything.

Some idea?

**Comment by nevenne** [comment]
error: function(xhr, status, error) {
  var err = eval("(" + xhr.responseText + ")");
  alert(err.Message);
}

**Comment by sabatesduran** [comment]
@nevenne If I write what you are saying the result is this:
```javascript
xhr = null
status = JSON_MALFORMED
error = null
```

I see the code on phonon.js and error function is this:
```javascript
if (typeof error === 'function') {
    window.setTimeout(function() {
    error(res, flagError, xhr);
}, 1);
```
Thanks for your comment

**Comment by Qathom** [comment]
Hi @sabatesduran,

Did you analyse the request and the response by using the in-browser web development tools?
On Chrome, you can press F12, then under "Network", you can see what the request contains, same for the server's response.

**Comment by sabatesduran** [comment]
Nevermind I didn't see this last night, my server is returning Error 500.

Sorry to bother you.

**Comment by bmcbride** [comment]
It looks like this might still be a bug. AJAX requests with `data` don't seem to be including the query parameters. See the example below:

```
phonon.ajax({
  method: "GET",
  url: "https://jsonplaceholder.typicode.com/posts",
  dataType: "json",
  data: {
    "userId": "1"
  },
  success: function(res, xhr) {
    console.log(res);
  },
  error: function(res, flagError, xhr) {
    phonon.alert(flagError, "AJAX Error", true, "OK");
  }
});
```
It returns 100 results, not filtered by userId as shown in the Chrom console screenshot below.

![image](https://user-images.githubusercontent.com/753187/34582659-acd78928-f162-11e7-9061-b9b6e9ff96cd.png)

Removing `data` from the request and adding the parameters directly to the URL works as expected:

![image](https://user-images.githubusercontent.com/753187/34582771-1d49b708-f163-11e7-93f8-ddca05276f20.png)

**Comment by qathom** [comment]
Hello @sabatesduran and @bmcbride,

Thank you for opening this issue.
GET requests with parameters have been fixed in the latest release (v1.4.9).
